File: packages/shared/events/draft.ts
import type { DomainEvent } from './base';

export interface DraftPickMadeEvent extends DomainEvent {
  type: 'draft.pick_made';
  sourceService: 'draft-service';
  contestId: string;
  draftSessionId: string;
  teamId: string;
  participantId: string;
  pickNumber: number;
  round: number;
  autoPicked: boolean;
}

export interface DraftCompletedEvent extends DomainEvent {
  type: 'draft.completed';
  sourceService: 'draft-service';
  contestId: string;
  draftSessionId: string;
  totalPicks: number;
}

File: packages/shared/poolmaster_shared/events/draft.py
"""Draft-related events."""

from uuid import UUID

from poolmaster_shared.events.base import DomainEvent


class DraftPickMadeEvent(DomainEvent):
    """Published when a pick is made during a draft."""

    type: str = "draft.pick_made"
    source_service: str = "draft-service"
    contest_id: UUID
    draft_session_id: UUID
    team_id: UUID
    participant_id: UUID
    pick_number: int
    round: int
    auto_picked: bool


class DraftCompletedEvent(DomainEvent):
    """Published when a draft finishes."""

    type: str = "draft.completed"
    source_service: str = "draft-service"
    contest_id: UUID
    draft_session_id: UUID
    total_picks: int
